### DISCUSSION TITLE
NFT are not getting shown.

### DISCUSSION THREAD

**Discussion by rudreshkoranne** [aptos-labs/aptos-developer-discussions#524] [Questions] [answered]
### Describe your question in detail.

Hello everyone,
I'm currently learning how to mint and transfer an NFT. I wrote a contract and deployed it at the address: 0xf8da3ca3ec059b47c5b7f68c9ba1c0279a1ad80923f032807bbaae0bd47459dd::poa_nft on DEVNET. The token is minting successfully, but it's not showing up in my Petra wallet or on the explorer.
Can anyone please help me with this?

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by CryptainGolD** [comment]
Let you run that action on testnet.

**Reply by rkoranne0755** [reply]
One more thing I want to clarify,
I wrote a smart contract to mint a soulbound NFT and minted few NFT's into my wallet address on TESTNET.
The NFT's are showning in my petra wallet but not on any of the Marketplace.
Can you please explain me why so?

**Reply by rkoranne0755** [reply]
Contract Address: 0x2a0abf2355b42068acea7a8b5862785bf3656f9225872e2020480ca2eaa3e66c
wallet address: 0x3fb90e96842b5d4035ab1eab45df216b235f3c495777672471523daa199eb188

**Reply by CryptainGolD** [reply]
You can browse your tokens in https://testnet.rarible.com
Navigate into Wallet / MyNFTs 👍

**Comment by gregnazario** [comment, marked as answer]
NFTs minted on devnet, will only show in the wallet or explorer on devnet.  Though, sometimes there is maintenance on the devnet indexer, which may cause those to not show.

Just ensure that you're matching the network with each of the tools

**Reply by rudreshkoranne** [reply]
Thnxx for the help!!!
